app/frontend/src/features/donations/DonationsBox.tsx

```diff
@@ -206,42 +202,39 @@ const useStyles = makeStyles((theme) => ({
 
 export interface DonationFormData {
   amount: number;
-  recurring: boolean;
+  recurring: "monthly" | "one-off";
 }
 
 export default function DonationsBoxMixed() {
   const stripePromise = loadStripe(process.env.REACT_APP_STRIPE_KEY!);
 
   const classes = useStyles();
 
-  const [success, setSuccess] = useState(false);
-  const [cancelled, setCancelled] = useState(false);
+  const [isPredefinedAmount, setisPredefinedAmount] = useState(false);
 
   const location = useLocation();
+  const query = new URLSearchParams(location.search);
+  const [success, setSuccess] = useState(query.get("success") ? true : false);
+  const [cancelled, setCancelled] = useState(query.get("cancelled") ? true : false);
 
-  const [isAmount, setIsAmount] = useState(false);
-
-  useEffect(() => {
-    const query = new URLSearchParams(location.search);
-    setSuccess(query.get("success") ? true : false);
-    setCancelled(query.get("cancelled") ? true : false);
-  }, [location]);
 
   const {
     control,
     handleSubmit,
     reset: resetForm,
     errors,
-  } = useForm<DonationFormData>({ defaultValues: { recurring: true } });
+  } = useForm<DonationFormData>();
+
+  const customAmountInput = useRef<HTMLInputElement | null>(null)
 
   const {
     error,
     isLoading,
     mutate: initiateDonation,
   } = useMutation<void, GrpcError, DonationFormData>(
-    async (formData) => {
+    async ({ amount, recurring }) => {
       const stripe = (await stripePromise)!;
-      const session_id = await service.donations.initiateDonation(formData);
+      const session_id = await service.donations.initiateDonation({ amount, recurring: recurring = "monthly" });
```

Comment:
You've accidentally done assignment instead of a boolean test `recurring === "monthly"`. In the form, we changed the type of `recurring` from a boolean to a "state string", but `initiateDonation` is still expecting a boolean.